aws/src/index.js

```diff
@@ -349,33 +348,54 @@ class S3HttpCollector extends Collector {
   }
 }
 
-/* eslint-disable no-param-reassign */
-const processLogText = (cloudWatchLogs, tagRegexMap, arn) => {
-  // derive source
-  const keys = arn.split(':');
-  const region = keys[3];
-  const accountId = keys[4];
+function deriveRegionAndAccount(arn){
+  let region;
+  let accountId;
 
-  // default value
-  let source = accountId + '-' + region;
-
-  // derive source for lambda function or EC2 instance
+  if (arn) {
+    const keys = arn.split(':');
+    region = keys[3];
+    accountId = keys[4];
+  }
+return {region, accountId};
+}
+function deriveSource(cloudWatchLogs,source){
   const logGroup = cloudWatchLogs.logGroup;
   const logStream = cloudWatchLogs.logStream;
 
-  if(logGroup.startsWith('/aws/lambda/')) {
+  if(logGroup && logGroup.startsWith('/aws/lambda/')) {
     const logGroupKeys = logGroup.split('/');
     source = logGroupKeys[logGroupKeys.length - 1];
-  } else if (logStream.startsWith('i-')) {
+  } else if (logStream && logStream.startsWith('i-')) {
     const logStreamKeys = logGroup.split('-');
     source = logStreamKeys[logStreamKeys.length - 1];
   }
+return source;
+}
+/* eslint-disable no-param-reassign */
+const processLogText = (cloudWatchLogs, tagRegexMap, arn) => {
+  // derive accountId & Region
+  const { region, accountId } = deriveRegionAndAccount(arn);
+
+  // derive source: default
+  let source = accountId + '-' + region;
```

Comment:
Minor: lets move this into deriveSource() method.

Reply:
Sure. Will do.